File: server/data/results/translations/anchored/go/two_sum.go
package main

import "fmt"

func two_sum(nums []int, target int) []int {
	seen := make(map[int]int)
	for i := 0; i < len(nums); i++ {
		complement := target - nums[i]
		if val, ok := seen[complement]; ok {
			return []int{val, i}
		} else {
			seen[nums[i]] = i
		}
	}
	return []int{-1, -1}
}

func main() {
	cases := []struct {
		nums   []int
		target int
	}{
		{[]int{2, 7, 11, 15}, 9},
		{[]int{3, 2, 4}, 6},
		{[]int{1, 2, 3}, 100},
	}
	for _, c := range cases {
		pair := two_sum(c.nums, c.target)
		fmt.Printf("%d %d\n", pair[0], pair[1])
	}
}

File: server/data/results/translations/anchored/javascript/two_sum.js
function two_sum(nums, target) {
    const seen = {};
    for (let i = 0; i < nums.length; i++) {
        const complement = target - nums[i];
        if (complement in seen) {
            return [seen[complement], i];
        } else {
            seen[nums[i]] = i;
        }
    }
    return [-1, -1];
}


function main() {
    const cases = [
        [[2, 7, 11, 15], 9],
        [[3, 2, 4], 6],
        [[1, 2, 3], 100]
    ];
    for (const [nums, target] of cases) {
        const pair = two_sum(nums, target);
        console.log(String(pair[0]) + " " + String(pair[1]));
    }
}


main();